src/Shadow.ts

```diff
@@ -0,0 +1,171 @@
+import { BLEND_MODES, SCALE_MODES } from '@pixi/constants';
+
+import { Application } from '@pixi/app';
+import { RenderTexture } from '@pixi/core';
+import { ShadowMapFilter } from './filters/ShadowMapFilter';
+import { ShadowMaskFilter } from './filters/ShadowMaskFilter';
+import { Sprite } from '@pixi/sprite';
+
+/**
+ * @class
+ * @memberof PIXI.shadows
+ *
+ * @param range {number} The radius of the lit area in pixels.
+ * @param [intensity=1] {number} The opacity of the lit area.
+ * @param [pointCount=20] {number} The number of points that makes up this light.
+ * @param [scatterRange=15] {number} The radius at which the points of the light should be scattered.
+ */
+
+export class Shadow extends Sprite {
+    /**
+     * The of steps to take per pixel. (Higher resolution = more precise edges + more intensive).
+     */
+    depthResolution = 1; // per screen pixel
+    /**
+     * Whther or not overlays in shadows should become darker (can create odd artifacts, is very experimental/unfinished)
+     */
+    darkenOverlay = false;
+
+    /**
+     * How many pixels of the overlay should be lit up by the light
+     */
+    overlayLightLength = Infinity;
+
+    /**
+     * A shadow caster to ignore while creating the shadows. (Can be used if sprite and light always overlap).
+     */
+    ignoreShadowCaster: Sprite | undefined;
+
+    shadowFilter: ShadowMaskFilter;
+    renderStep: boolean | undefined;
+    shadowMapResultSprite: Sprite | undefined;
+
+    _shadowCasterSprite: Sprite | undefined;
```

Comment:
- Everything that is not accessed from the outside is marked as `private`, optional parameters that no one ever uses are no longer parameters, etc. Narrowing the API surface is a good practice IMO. However the API surface is still pretty large, mostly IIRC to support some customizations in the demos. 
 -  names were left as they were, because so much is changed already here. 
 - public getters and setters around a private field with no additional logic at all were removed, and the enclosed property was renamed and made public to not break compatibility. I felt they added clutter and served no purpose.